### PR TITLE
add quill

### PR DESCRIPTION
**Pull request by star-hengxing**
Asynchronous Low Latency C++ Logging Library.

### CODE REVIEW

File: packages/q/quill/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+             "https://github.com/odygrd/quill.git")
+
+    add_versions("2.8.0", "0461a6c314e3d882f3b9ada487ef1bf558925272509ee41a9fd25f7776db6075")
+
+    add_configs("fmt_external", {description = "Use external fmt library instead of bundled.", default = false, type = "boolean"})
+    add_configs("noexcept", {description = "Build without exceptions with -fno-exceptions flag", default = false, type = "boolean"})
+
+    add_deps("cmake")
+
+    on_load("windows", "linux", "macosx", function (package)
```
Comment (waruqi) [inline packages/q/quill/xmake.lua:16]: 只需要 on_install 限制下就行了，这里没必要加平台，

**Comment by star-hengxing** [comment]
@waruqi

**Comment by evanzd** [comment]
@star-hengxing Thanks for your adding quill 👍. It seems the last `on_test` step will fail on ubuntu (maybe need to link pthread?). Do you have any idea how to fix this or know any workaround to skip call `on_test`?

error info below:

```
/usr/bin/ld: /home/user/.xmake/packages/q/quill/2.8.0/aa2a24c05cf94f6aba798a141ecc5ce6/lib/libquill.a(Os.cpp.o): in function `quill::detail::get_thread_name[abi:cxx11]()':
Os.cpp:(.text.unlikely+0x1ad): undefined reference to `pthread_getname_np'
/usr/bin/ld: /tmp/.xmake1005/230531/_0CE7529E3B874C208D4351EE9A644740.o: in function `std::thread::thread<quill::detail::BackendWorker::run()::{lambda()#1}, , void>(quill::detail::BackendWorker::run()::{lambda()#1}&&)':
_AEF1856914024A4DB4184B66737B32D8.cpp:(.text._ZNSt6threadC2IZN5quill6detail13BackendWorker3runEvEUlvE_JEvEEOT_DpOT0_[_ZNSt6threadC5IZN5quill6detail13BackendWorker3runEvEUlvE_JEvEEOT_DpOT0_]+0x33): undefined reference to `pthread_create'
collect2: error: ld returned 1 exit status
```

**Comment by waruqi** [comment]
> @star-hengxing Thanks for adding quill 👍. It seems the last `on_test` step will fail on ubuntu (maybe need to link pthread?). Do you have any idea how to fix this or know any workaround to skip call `on_test`?
> 
> error info below:
> 
> ```
> /usr/bin/ld: /home/user/.xmake/packages/q/quill/2.8.0/aa2a24c05cf94f6aba798a141ecc5ce6/lib/libquill.a(Os.cpp.o): in function `quill::detail::get_thread_name[abi:cxx11]()':
> Os.cpp:(.text.unlikely+0x1ad): undefined reference to `pthread_getname_np'
> /usr/bin/ld: /tmp/.xmake1005/230531/_0CE7529E3B874C208D4351EE9A644740.o: in function `std::thread::thread<quill::detail::BackendWorker::run()::{lambda()#1}, , void>(quill::detail::BackendWorker::run()::{lambda()#1}&&)':
> _AEF1856914024A4DB4184B66737B32D8.cpp:(.text._ZNSt6threadC2IZN5quill6detail13BackendWorker3runEvEUlvE_JEvEEOT_DpOT0_[_ZNSt6threadC5IZN5quill6detail13BackendWorker3runEvEUlvE_JEvEEOT_DpOT0_]+0x33): undefined reference to `pthread_create'
> collect2: error: ld returned 1 exit status
> ```

you can add it. https://github.com/xmake-io/xmake-repo/blob/6a77f128b7f8620b1f62fdc5434c80277a3e7f27/packages/p/pthreadpool/xmake.lua#L16

**Comment by evanzd** [comment]
> > @star-hengxing Thanks for adding quill 👍. It seems the last `on_test` step will fail on ubuntu (maybe need to link pthread?). Do you have any idea how to fix this or know any workaround to skip call `on_test`?
> > error info below:
> > ```
> > /usr/bin/ld: /home/user/.xmake/packages/q/quill/2.8.0/aa2a24c05cf94f6aba798a141ecc5ce6/lib/libquill.a(Os.cpp.o): in function `quill::detail::get_thread_name[abi:cxx11]()':
> > Os.cpp:(.text.unlikely+0x1ad): undefined reference to `pthread_getname_np'
> > /usr/bin/ld: /tmp/.xmake1005/230531/_0CE7529E3B874C208D4351EE9A644740.o: in function `std::thread::thread<quill::detail::BackendWorker::run()::{lambda()#1}, , void>(quill::detail::BackendWorker::run()::{lambda()#1}&&)':
> > _AEF1856914024A4DB4184B66737B32D8.cpp:(.text._ZNSt6threadC2IZN5quill6detail13BackendWorker3runEvEUlvE_JEvEEOT_DpOT0_[_ZNSt6threadC5IZN5quill6detail13BackendWorker3runEvEUlvE_JEvEEOT_DpOT0_]+0x33): undefined reference to `pthread_create'
> > collect2: error: ld returned 1 exit status
> > ```
> 
> you can add it.
> 
> https://github.com/xmake-io/xmake-repo/blob/6a77f128b7f8620b1f62fdc5434c80277a3e7f27/packages/p/pthreadpool/xmake.lua#L16

`add_syslinks("pthread")` not working. Is there any example to link `pthread` in `on_test`?

image
image

**Comment by waruqi** [comment]
image

**Pull request merged**